routers/web/repo/search.go

```diff
@@ -42,24 +39,51 @@ func Search(ctx *context.Context) {
 		page = 1
 	}
 
-	total, searchResults, searchResultLanguages, err := code_indexer.PerformSearch(ctx, &code_indexer.SearchOptions{
-		RepoIDs:        []int64{ctx.Repo.Repository.ID},
-		Keyword:        keyword,
-		IsKeywordFuzzy: isFuzzy,
-		Language:       language,
-		Paginator: &db.ListOptions{
-			Page:     page,
-			PageSize: setting.UI.RepoSearchPagingNum,
-		},
-	})
-	if err != nil {
-		if code_indexer.IsAvailable(ctx) {
-			ctx.ServerError("SearchResults", err)
-			return
+	var total int
+	var searchResults []*code_indexer.Result
+	var searchResultLanguages []*code_indexer.SearchResultLanguages
+	if setting.Indexer.RepoIndexerEnabled {
```

Comment:
I would suggest to intigrate the git grep search as its own indexer and set it as default.

This way it is transparent for webUI or API what to do.

Reply:
No idea how to do that clearly, and I am not a fan of adding a lot of "options".

If you have better ideas, free free to edit this PR directly or have some following PRs.

Reply:
I added a prompt like this, maybe it could make it clearer. What do you think?

<details>

![image](https://github.com/go-gitea/gitea/assets/2114189/45f737d2-1c8d-42bc-af89-eb21efee0102)

</details>

Reply:
that will help but is unrelated to the architecture idea of mine.

I try to create a pull request to your branch that would move acording to my proposal, so it can be checked out and tested etc ...

Reply:
I would expect that there is no new option to be introduced.